include/range/v3/detail/config.hpp

```diff
@@ -757,4 +757,11 @@ namespace ranges
 #endif
 #endif
 
+#if defined(__has_include) && __has_include(<span>) && \
+    defined(__cpp_lib_span) && __cpp_lib_span > 0
```

Comment:
Not all stdlib implementations have a `<version>` header. Are there any stdlib implementations have a usable `<span>` header but not a `<version>` header (plausable), or that have a `<version>` header but fail to define `__cpp_lib_span` in it (unlikely)? If so, this risks concluding that there is no usable `<span>` header when in fact there might be.

Reply:
This is a good point. The problematic case is, as you said, a compiler that has a usable `<span>` header (which defines `__cpp_lib_span`) but no `<version>` header. Which means the only way to know it's usable would be to include it.

I think a good approach would be to blacklist the known problematic case. In this case: a workaround macro that is defined when compiling with MSVC and C++17 (or lower), and then avoid including `<span>` if the macro it's defined.

I have force-pushed a new implementation based on that, and edited the PR description to match